drivers/flash/spi_nor.c

```diff
@@ -821,8 +837,17 @@ static int spi_nor_erase(const struct device *dev, off_t addr, size_t size)
 static int spi_nor_write_protection_set(const struct device *dev,
 					bool write_protect)
 {
+#if ANY_INST_HAS_WP_GPIOS
+	const struct spi_nor_config *cfg = dev->config;
+#endif
 	int ret;
 
+#if ANY_INST_HAS_WP_GPIOS
+	if (cfg->wp) {
+		gpio_pin_set_dt(cfg->wp, write_protect);
+	}
+#endif
+
```

Comment:
We will need to figure this out, probably later, but right now it seems that we are placing two write locks for, probably, no reason.

Reply:
@de-nordic If I understand correctly, the driver presently supports soft write-protect and this PR adds GPIO write-protect.  

Users that tie WP disabled in HW won't define wp-gpios, so this PR should not change anything for them.  Users that connect WP to their MCU and then define wp-gpios in DTS incorrectly assume it's doing something (as far as I can tell it doesn't even initialize) and this PR addresses that.

Perhaps we should consider that defining wp-gpios in the DTS should disable the existing soft write-protect in this driver in favor of the gpio write-protect?  Personally I am OK with redundant write-protect!

Reply:
Twice the reasons to have errors and extra code, but let it be as is.

Reply:
I too dislike code!